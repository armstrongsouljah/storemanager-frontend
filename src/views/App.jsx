/* eslint-disable react/jsx-indent */
/* eslint-disable no-shadow */
import React, { Component, } from "react";
import { Redirect, } from "react-router-dom";
import { connect, } from "react-redux";
import LoginForm from "../components/loginForm";
import loginAction from "../actions/loginAction";

export class App extends Component {
    state = {
        username: "",
        password: "",
        redirect: false,
        loading: false,
    }

    componentWillReceiveProps(props) {
        this.handleIsAuthenticated(props.loginState);
    }

    handleIsAuthenticated = (loginState) => {
        if (loginState.token) {
            this.setState({ redirect: true, });
        }
    }

    handleSubmit = (event) => {
        event.preventDefault();
        const { loginAction, } = this.props;
        loginAction(this.state);
        this.setState({ loading: true, });
    }

    onChange = (event) => {
        event.preventDefault();
        this.setState({ [event.target.id]: event.target.value, });
    }

    render() {
        const { redirect, } = this.state;
        const { errors, } = this.props.loginState;
        if (redirect) {
            return (
                <Redirect to="/dashboard" />
            );
        } return (
            <LoginForm onSubmit={this.handleSubmit} onChange={this.onChange} error={errors} />
        );
    }
}

const mapStateToProps = state => ({
    loginState: state.loginReducer,
});

export default connect(mapStateToProps, { loginAction, })(App);
