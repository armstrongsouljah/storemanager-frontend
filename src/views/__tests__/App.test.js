/* eslint-disable import/no-named-as-default */
import React from "react";
import { shallow, } from "enzyme";
import App, { App as App2, } from "../App";

const props = {
    loginState: {
        userRole: null,
        token: null,
        successMessage: null,
        errors: null,
    },
    loginAction: jest.fn(),
};

describe("App", () => {
    it("renders without crashing", () => {
        const wraper = new App(props);
        expect(wraper).toMatchSnapshot();
    });
    it("should set state onChange", () => {
        const wrapper = shallow(<App2 {...props} />);
        wrapper.instance().setState = jest.fn();
        const event = {
            target: {
                id: "",
            },
            preventDefault: jest.fn(),
        };
        wrapper.instance().onChange(event);
        expect(wrapper.instance().setState).toBeCalled();
    });
    it("should call loginAction on handleSubmit", () => {
        const wrapper = shallow(<App2 {...props} />);
        wrapper.instance().handleSubmit({ preventDefault() {}, });
        expect(wrapper.instance().props.loginAction).toBeCalled();
    });
    it("should handleIsAuthenticated on component will receive props", () => {
        const instance = new App2(props);
        instance.handleIsAuthenticated = jest.fn();
        instance.setState = jest.fn();
        instance.componentWillReceiveProps(props);
        expect(instance.handleIsAuthenticated).toBeCalled();
    });
});
