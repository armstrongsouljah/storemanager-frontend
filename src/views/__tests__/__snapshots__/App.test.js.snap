// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
Connect {
  "context": undefined,
  "indirectRenderWrappedComponent": [Function],
  "props": Object {
    "loginAction": [MockFunction],
    "loginState": Object {
      "errors": null,
      "successMessage": null,
      "token": null,
      "userRole": null,
    },
  },
  "refs": Object {},
  "selectChildElement": [Function],
  "selectDerivedProps": [Function],
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
